demos/benchmarks/README.md

```diff
@@ -0,0 +1,86 @@
+# PowerSync Benchmark app
+
+This is an app to test:
+1. Initial sync time. This can be used to get a sync throughput estimate.
+2. Incremental sync latency.
+
+Incremental sync latency tests the total latency of the following sequence:
+
+1. Create a row on the client, recording the creation time.
+2. Upload to server via NodeJS demo backend.
+3. Postgres adds a default value to one column.
+4. The row is synced back down, including the new default column.
+5. The client detects that the default column is set (using a watched query).
+6. The client updates the row, recording the latency.
+
+For initial sync, bulk data is synced without actively being used by the app, just to measure the sync time.
+
+# Setup
+
+To primarily measure the client-side sync overhead, run the PowerSync service on the same local network as the client.
+
+## Postgres
+
+```sql
+CREATE TABLE bulk_data (id uuid primary key default gen_random_uuid(), created_at timestamptz not null default now(), name text, size_bucket text);
+CREATE TABLE benchmark_items(id uuid primary key, description text, client_created_at timestamptz not null, client_received_at timestamptz, server_created_at timestamptz not null default now());
+
+INSERT INTO bulk_data (name, size_bucket) SELECT repeat('a', 20), '10k' FROM generate_series(1, 10000);
+INSERT INTO bulk_data (name, size_bucket) SELECT repeat('a', 20), '100k' FROM generate_series(1, 100000);
+INSERT INTO bulk_data (name, size_bucket) SELECT repeat('a', 20), '1m' FROM generate_series(1, 1000000);
+INSERT INTO bulk_data (name, size_bucket) SELECT repeat('a', 20), '10m' FROM generate_series(1, 10000000);
```

Comment:
Should we add a statement for ensuring the publication exists?

Reply:
In general my goal here isn't to provide the full setup instructions for a powersync instance, but rather just document the bits that are specific to the benchmarks.